.github/workflows/master.yml

```
@@ -35,6 +38,28 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
           args: --all-features -- -D warnings
 
+  solana-build: 
+    name: Cargo Build using bpf (solana env)
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/checkout@v3
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          components: rustfmt
+
+      - name: Install Solana CLI
+        run: curl -sSfL https://release.solana.com/$SOLANA_VERSION/install | sh
+        shell: bash
+
+      - name: Adding Solana CLI to the path
+        run: echo "/home/runner/.local/share/solana/install/active_release/bin" >> $GITHUB_PATH
+        shell: bash
```

Probably just me but I find the single `echo` instruction as separate
step triggering. ;)  Maybe join them?

```suggestion
      - name: Install Solana CLI
        run: |
          set -eux
          curl -sSfL https://release.solana.com/$SOLANA_VERSION/install | sh
          echo "$HOME/.local/share/solana/install/active_release/bin" >> $GITHUB_PATH
```

Also note that `bash` is the default shell on Linux so no need to specify it.

---

alright understood, lemme try that.

---

looks like the above didnt work.